File: app/src/main/java/cn/teach/equip/view/SplashActivity.java
package cn.teach.equip.view;

import android.os.Bundle;
import android.os.Handler;
import android.support.annotation.Nullable;

import com.blankj.utilcode.util.StringUtils;

import cn.teach.equip.R;
import cn.teach.equip.api.HttpResultSubscriber;
import cn.teach.equip.api.HttpServerImpl;
import cn.teach.equip.base.BaseActivity;
import cn.teach.equip.base.MyApplication;
import cn.teach.equip.bean.pojo.UserBO;
import cn.teach.equip.view.main.MainActivity;
import cn.teach.equip.view.splash.SplashPage1;

public class SplashActivity extends BaseActivity {

    @Override
    protected int getLayout() {
        return R.layout.act_splash;
    }


    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                boolean isFirst = MyApplication.spUtils.getBoolean("isFirst", true);
                if (!isFirst) {
                    String token = MyApplication.spUtils.getString("token");
                    if (StringUtils.isEmpty(token)) {
                        gotoActivity(MainActivity.class, true);
                    } else {
                        MyApplication.token = token;
                        login(null, null);
                    }
                } else {
                    MyApplication.spUtils.put("isFirst", false);
                    gotoActivity(SplashPage1.class, true);
                }
            }
        }, 3000);
    }


    public void login(String phone, String smsCode) {
        HttpServerImpl.login(phone, smsCode).subscribe(new HttpResultSubscriber<UserBO>() {
            @Override
            public void onSuccess(UserBO s) {
                MyApplication.token = s.getUserToken();
                MyApplication.userBO = s;
                MyApplication.spUtils.put("token", MyApplication.token);
                gotoActivity(MainActivity.class, true);
            }

            @Override
            public void onFiled(String message) {
                MyApplication.token = null;
                gotoActivity(MainActivity.class, true);
            }
        });
    }
}

File: app/src/main/java/cn/teach/equip/view/SplashActivity.kt
package cn.teach.equip.view

import android.os.Bundle
import android.os.Handler
import cn.teach.equip.R
import cn.teach.equip.api.HttpResultSubscriber
import cn.teach.equip.api.HttpServerImpl
import cn.teach.equip.base.BaseActivity
import cn.teach.equip.base.MyApplication
import cn.teach.equip.bean.pojo.UserBO
import cn.teach.equip.view.main.MainActivity
import cn.teach.equip.view.splash.SplashPage1
import com.blankj.utilcode.util.StringUtils

class SplashActivity : BaseActivity() {
    override fun getLayout(): Int {
        return R.layout.act_splash
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Handler().postDelayed({
            val isFirst = MyApplication.spUtils.getBoolean("isFirst", true)
            if (!isFirst) {
                val token = MyApplication.spUtils.getString("token")
                if (StringUtils.isEmpty(token)) {
                    gotoActivity(MainActivity::class.java, true)
                } else {
                    MyApplication.token = token
                    login(null, null)
                }
            } else {
                MyApplication.spUtils.put("isFirst", false)
                gotoActivity(SplashPage1::class.java, true)
            }
        }, 3000)
    }

    fun login(phone: String?, smsCode: String?) {
        HttpServerImpl.login(phone, smsCode).subscribe(object : HttpResultSubscriber<UserBO?>() {
            override fun onSuccess(t: UserBO?) {
                MyApplication.token = t?.userToken
                MyApplication.userBO = t
                MyApplication.spUtils.put("token", MyApplication.token)
                gotoActivity(MainActivity::class.java, true)
            }

            override fun onFiled(message: String) {
                MyApplication.token = null
                gotoActivity(MainActivity::class.java, true)
            }
        })
    }
}